hanadb_exporter/exporters/prometheus_metrics.py

```diff
@@ -8,22 +8,86 @@
 :since: 2019-05-09
 """
 
+import logging
+import collections
 import json
 
 
+METRICMODEL = collections.namedtuple(
+    'Metric',
+    'name description labels value unit type enabled'
+)
+
+
+class Metric(METRICMODEL):
+    """
+    Class to store the loaded prometheus metrics inherited from namedtuple
+    """
+
+    # pylint:disable=R0913
+    # pylint:disable=W0622
+    def __new__(cls, name, description, labels, value, unit, type, enabled=True):
+        return super(Metric, cls).__new__(
+            cls, name, description, labels, value, unit, type, enabled)
+
+
+class Query(object):
+    """
+    Class to store the query and its metrics
+    """
+
+    def __init__(self):
+        self.query = None
+        self.metrics = []
+        self.enabled = True
+
+    def parse(self, query, query_data):
+        """
+        Parse metrics by query
+        """
+        # TODO: Could we add a simple SQL query syntax check?
```

Comment:
@arbulu89 AFAIK the easiest and more secure way to do it is using the database parser that is executed during the opening the cursor. As this has a high cost for the database, I think it makes sense to treat the error when executing and providing a proper log of the Syntax error. According to with the specification, we should intercept the `ProgrammingError` exception.

https://legacy.python.org/dev/peps/pep-0249/#cursor-methods